File: 프로그래머스/0/120890. 가까운 수/가까운 수.js
function solution(array, n) {
    var answer = 0;
    
    array.sort((a, b) => {
        if(Math.abs(a - n) === Math.abs(b - n))
            return a - b;
        else
            return Math.abs(a - n) - Math.abs(b - n);
    })
    
    return array[0];
}

File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
import java.util.Arrays;

class Solution {
    public int solution(int[] array, int n) {
        return Arrays.stream(array)
            .boxed()
            .sorted((a, b) -> {
                int diffA = Math.abs(a - n);
                int diffB = Math.abs(b - n);
                if (diffA == diffB) {
                    return a.compareTo(b);  // 차이가 같을 경우 작은 수를 먼저
                }
                return Integer.compare(diffA, diffB);
            })
            .findFirst()
            .orElse(0);
    }
}
